index.js

```diff
@@ -56,12 +56,12 @@ function interpolate(t, degree, points, knots, weights, result) {
     v[i][d] = weights[i];
   }
 
-  // l (level) goes from 1 to the curve degree
+  // l (level) goes from 1 to the curve degree + 1
   var alpha;
-  for(l=1; l<=degree; l++) {
+  for(l=1; l<=degree+1; l++) {
     // build level l of the pyramid
-    for(i=s; i>s-degree+l; i--) {
-      alpha = (t - knots[i]) / (knots[i+degree-l] - knots[i]);
+    for(i=s; i>s-degree-1+l; i--) {
```

Comment:
In the meantime I'll push the new major as soon as possible. In case you need it for any reason feel free to push / publish it before I do. I'm buried under a lot of work these days so it may take a few.